Source/Meadow.Contracts/Hardware/Networking/INetworkAdapterCollection.cs

```diff
@@ -1,9 +1,15 @@
 ﻿using System.Collections.Generic;
+using System.Linq;
 
 namespace Meadow.Hardware
 {
     public interface INetworkAdapterCollection : IEnumerable<INetworkAdapter>
     {
         INetworkAdapter this[int index] { get; }
+
+        public T Primary<T>() where T : INetworkAdapter
```

Comment:
Just a note, if we use `Primary` over `First`; it has semantic meaning, in that we should make sure that the underlying call to `First()` actually returns whatever "Primary" NIC we want to setup. If it's truly just a call to the `First()` adapter, we might consider renaming it to `First`. Personally, I like `Primary` or `Default`, but it implies prescription.

Reply:
It's in the interface and if we ever implement something with > 1 WiFi or > 1 wired, we can deal with it then.